hy/compiler.py

```diff
@@ -2689,18 +2689,20 @@ def hy_compile(tree, module_name, root=ast.Module, get_expr=False):
     body = []
     expr = None
 
-    if tree:
+    if not (isinstance(tree, HyObject) or type(tree) is list):
```

Comment:
Also, this is an internal function. If it receives an argument that's of a type we didn't expect, that's a bug.